CONTRIBUTING.rst

```diff
@@ -0,0 +1,33 @@
+Contributing to MNE-Python
+==========================
+
+MNE-Python is maintained by a community of scientists and research labs, and
+accepts contributions in the form of bug reports, fixes, feature additions, and
+documentation improvements (even just typo corrections). The best way to start
+contributing is by `opening an issue`_ on our GitHub page to tell us about
+errors you are encountering or to discuss your ideas for changes. For general
+troubleshooting, you can also write to the `MNE mailing list`_ or chat with
+developers on the `MNE gitter channel`_.
+
+Changes are typically made by `forking`_ the MNE-Python repository, making
+changes to your fork (usually by `cloning`_ it to your personal computer,
+making the changes, and then `pushing`_ the local changes up to your fork), and
+finally creating a `pull request`_ to incorporate your changes back into the
+shared "upstream" version of the codebase. The documentation section
+":ref:`dev-setup`" has more details on how best to configure your workflow for
+a smooth experience contributing to MNE-Python.
```

Comment:
Where is ":ref:`dev-setup`" pointing to? I personally would have found it very helpful to have the complete git-workflow to contribute in a very short version in one place. Just a list of commands with short comments in-between. I could put together what I ended up to do as a starting point, if that would help.
(i.e.: fork, clone, checkout master, [rebase to upstream], branch, commit (with a template for commit messasges, e.g. "FIX: .../DOC: ..."), push -u, PR, [pull, commit, push], delete/prune branch locally and remotely; or alternatively merge locally). By getting everything from various sources, I made quite a few mistakes underway.

Reply:
> I could put together what I ended up to do as a starting point, if that would help.

That would be a nice follow-up PR, basically a step-by-step example for how you could do some particular PR. Maybe show how you could have from scratch done the examplar PR (`mne-model-pr`) that @drammock and @massich came up with.

Reply:
@jeythekey good catch, the :ref:`dev-setup: link has been replaced by a proper URL (again, because we want `CONTRIBUTING.rst` to make sense without being rendered.  Regarding the basic git workflow, that section hadn't been finished yet when I pushed the first commit (hence the `WIP` title).  It's there now, here: https://12938-1301584-gh.circle-artifacts.com/0/dev/contributing.html#basic-git-commands

Reply:
@drammock , great, that is very nicely written and clear, would have helped me a lot, thanks.